### PR TITLE
Task/implement soft delete for all the models/1206183600212419

### PR DESCRIPTION
**Pull request by Ycandido0119**
### Asana Ticket

_Implement soft delete for all the models_

### PR Description

 - Add a deleted field to the `Meter`,`MeterPoint` and `MeterRegister` models
 - Make a migration to implement changes
 - Amend the `views.py` file to showcase soft deletion for all models
 - Add testing, also refactor testing page

### Checklist

- [ x ] Went through and checked that I haven't commited anything I shouldn't.
- [ x ] I have gone over the commit with a mentor.
- [ x ] All tests are passing
- [ ] I've manually verified the acceptance criteria is met
- [ ] Any relevant documentation is updated

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,30 +2,41 @@ asgiref==3.7.2
...
 pytest==8.0.0
 pytest-django==4.8.0
+PyYAML==6.0.1
+regex==2023.12.25
 ruff==0.1.13
-setuptools==69.0.3
 six==1.16.0
 sqlparse==0.4.4
 stack-data==0.6.3
+tqdm==4.66.2
```
Comment (todaatsushi) [inline requirements.txt:40]: What's up with the changes to the requirements?

Comment (Ycandido0119) [reply requirements.txt:40]: I think when I ran black it must have formatted the requirements file, if you want I can undo that commit.

**Review by todaatsushi — APPROVED** [review]

**Comment by todaatsushi** [comment]
Approved pending the rollback of requirements.txt changes

File: requirements.txt (hunk 1 above)
Comment (todaatsushi) [reply requirements.txt:40]: approved, but let's roll this back first

**Pull request merged**